File: programmers/Level 2/카펫.py
def solution(brown, yellow):
    answer = []
    total = brown + yellow

    for x in range(1, yellow + 1):
        if yellow % x == 0:
            y = yellow // x
            if x * y == yellow:
                data = (x + 2) * (y + 2) - brown
                if yellow == data:
                    answer = [x + 2, y + 2]
                    break
    answer.sort(reverse=True)
    return answer

File: programmers/skill Check 2/카펫.cpp
#include <string>
#include <vector>
#include <iostream>

using namespace std;

vector<int> solution(int brown, int yellow) {
    vector<int> answer;
    
    int total = brown + yellow;
    int left, right;
    
    for(int i =3; i<total; i++){
        if (total % i == 0){
            left = total / i;
            right = i;
            
            if (left>= right){
                
                int yellow_count = (left - 2) * (right - 2);
                
                if (yellow_count == yellow){
                    answer.push_back(left);
                    answer.push_back(right);
                    return answer;
                }
                
            }
            
        }
    }
    
    return answer;
}